src/local.rs

```diff
@@ -98,6 +98,19 @@ impl LocalAsset {
         }
     }
 
+    /// Writes an asset to a path on the local filesystem, determines the
+    /// filename from the origin path
```

Comment:
What does "determines filename from the origin path" mean?

Reply:
an asset knows where it came from (origin path) and what it contains- so if you make a new asset and go to write it- it needs to figure out what it's filename is. local assets do this from the origin_path exclusively. remote assets it's a combination of the url and the mimetype (because sometimes urls don't have filenames)